component/custom/custom_loader.go

```diff
@@ -0,0 +1,40 @@
+package custom
+
+import (
+	"os"
+	"plugin"
+	"strings"
+)
+
+func init() {
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		return
+	}
+
+	// environment variable can contain multiple paths separated by colon.
+	paths := strings.Split(pluginPath, ":")
+
+	for _, file := range paths {
+		println("custom_loader::Opening plugin:", file)
+		p, err := plugin.Open(file)
```

Comment:
This expects the file to be of *.so. You should filter the path to list only *.so files